libraries/node-core-library/src/Resolve.ts

```diff
@@ -3,15 +3,23 @@
 
 import { sync } from 'resolve';
 import * as path from 'path';
-import { FileSystem, PackageJsonLookup } from '@rushstack/node-core-library';
+
+import { PackageJsonLookup } from './PackageJsonLookup';
+import { FileSystem } from './FileSystem';
 
 const packageJsonLookup: PackageJsonLookup = new PackageJsonLookup();
```

Comment:
Should `PackageJsonLookup` have a default instance?  

The reason for not being a singleton is that if the filesystem changes after a lookup, the cache will return the old value for any subsequent lookups. This might be confusing for a task that is writing package.json files as outputs. 

But if a bunch of modules are all creating their own private caches, it defeats the point of caching.

Reply:
That's a good point. There's already a `clearCache()` function on it, and we actually already do have a static instance used by `PackageJsonLookup.loadOwnPackageJson`. What would you want a default singleton to look like?

Reply:
`PackageJsonLookup.instance`?

With a comment something like this:
```ts
/**
 * A singleton instance of `PackageJsonLookup`, which is useful for short-lived processes
 * that can reasonably assume that the file system will not be modified after the cache
 * is populated.
 *
 * @remarks
 * For long-running processes that need to clear the cache at appropriate times,
 * it is recommended to create your own instance of `PackageJsonLookup` instead
 * of relying on this instance.
 */
```